README.md

```diff
@@ -1,349 +1,152 @@
-  name: storage
-  namespace: openstack
-spec:
-  config: |
-    {
-      "cniVersion": "0.3.1",
-      "name": "storage",
-      "type": "macvlan",
-      "master": "enp7s0.21",
-      "ipam": {
-        "type": "whereabouts",
-        "range": "172.18.0.0/24",
-        "range_start": "172.18.0.50",
-        "range_end": "172.18.0.100"
-      }
-    }
+OPENSTACK_CR=~/openstack-deployment.yaml make openstack_deploy
 ```
 
-The following represents an example of Cinder resource that can be used
-to trigger the service deployment, and have the Cinder Volume and Cinder Backup
-service pods attached to the storage network using the above NetworkAttachmentDefinition.
+This will reboot the OpenShift cluster, so it will take a while to be up and
+running.
 
-```
-apiVersion: cinder.openstack.org/v1beta1
-kind: Cinder
-metadata:
-  name: cinder
-spec:
-  ...
-  cinderBackup:
-    ...
-    networkAttachents:
-    - storage
-  cinderVolume:
-    volume1:
-      ...
-      networkAttachents:
-      - storage
-...
+```sh
+oc wait --for condition=Ready --timeout=300s
 ```
 
-When the service is up and running, it will now have an additional nic
-configured for the storage network:
+Create the EDPM (compute) VM and provisioning:
 
-```
-# oc rsh cinder-volume-volume1-0
-Defaulted container "cinder-volume" out of: cinder-volume, probe, init (init)
-sh-5.1# ip a
-1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
-    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
-    inet 127.0.0.1/8 scope host lo
-       valid_lft forever preferred_lft forever
-    inet6 ::1/128 scope host
-       valid_lft forever preferred_lft forever
-3: eth0@if334: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue state UP group default
-    link/ether 0a:58:0a:82:01:44 brd ff:ff:ff:ff:ff:ff link-netns 7ea23955-d990-449d-81f9-fd57cb710daa
-    inet 10.130.1.68/23 brd 10.130.1.255 scope global eth0
-       valid_lft forever preferred_lft forever
-    inet6 fe80::bc92:5cff:fef3:2e27/64 scope link
-       valid_lft forever preferred_lft forever
-4: net1@if17: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
-    link/ether 26:35:09:03:71:b3 brd ff:ff:ff:ff:ff:ff link-netns 7ea23955-d990-449d-81f9-fd57cb710daa
-    inet 172.18.0.20/24 brd 172.18.0.255 scope global net1
-       valid_lft forever preferred_lft forever
-    inet6 fe80::2435:9ff:fe03:71b3/64 scope link
-       valid_lft forever preferred_lft forever
+```sh
+cd devsetup
+make edpm_compute
+
+cd ..
+DATAPLANE_TIMEOUT=40m make edpm_wait_deploy
 ```
 
-## Example: expose Cinder API to an isolated network
+We should now have a working OpenStack deployment and we will be able to run
+commands using the `openstackclient` pod:
 
-The Cinder spec can be used to configure Cinder API to register e.g.
-the internal endpoint to an isolated network. MetalLB is used for this
-scenario.
+```sh
+oc exec -t openstackclient -- openstack volume service list
```

Comment:
I think I asked this once before, but why use "oc exec -t" instead of "oc rsh"?

Reply:
I use `oc exec -t` when running just one command, and `rsh` to keep a remote shell open.

The `exec` command is a standard command that's available in all container engines (docker, podman, cri-o), so everybody is familiar with it.